GetVisibleServicesByScope/__tests__/handler.test.ts

```diff
@@ -0,0 +1,113 @@
+
+import {
+  NewService,
+  RetrievedService,
+  Service,
+  toAuthorizedCIDRs,
+  toAuthorizedRecipients
+} from "io-functions-commons/dist/src/models/service";
+import {
+  VISIBLE_SERVICE_BLOB_ID,
+  VISIBLE_SERVICE_CONTAINER,
+  VisibleService
+} from "io-functions-commons/dist/src/models/visible_service";
+
+import { MaxAllowedPaymentAmount } from "io-functions-commons/dist/generated/definitions/MaxAllowedPaymentAmount";
+
+import { aCosmosResourceMetadata } from "../../__mocks__/mocks";
+import {
+  GetVisibleServicesByScope,
+  GetVisibleServicesByScopeHandler
+} from "../handler";
+
+afterEach(() => {
+  jest.resetAllMocks();
+  jest.restoreAllMocks();
+});
+
+const anOrganizationFiscalCode = "01234567890" as OrganizationFiscalCode;
+
+const aService: Service = {
+  authorizedCIDRs: toAuthorizedCIDRs([]),
+  authorizedRecipients: toAuthorizedRecipients([]),
+  departmentName: "MyDeptName" as NonEmptyString,
+  isVisible: true,
+  maxAllowedPaymentAmount: 0 as MaxAllowedPaymentAmount,
+  organizationFiscalCode: anOrganizationFiscalCode,
+  organizationName: "MyOrgName" as NonEmptyString,
+  requireSecureChannels: false,
+  serviceId: "MySubscriptionId" as NonEmptyString,
+  serviceName: "MyServiceName" as NonEmptyString
+};
+
+const aNewService: NewService = {
+  ...aService,
+  kind: "INewService"
+};
+
+const aRetrievedService: RetrievedService = {
+  ...aNewService,
+  ...aCosmosResourceMetadata,
+  id: "123" as NonEmptyString,
+  kind: "IRetrievedService",
+  version: 1 as NonNegativeInteger
+};
+
+const aVisibleService: VisibleService = {
+  ...aCosmosResourceMetadata,
+  departmentName: aRetrievedService.departmentName,
+  id: aRetrievedService.id,
+  organizationFiscalCode: aRetrievedService.organizationFiscalCode,
+  organizationName: aRetrievedService.organizationName,
+  requireSecureChannels: false,
+  serviceId: aRetrievedService.serviceId,
+  serviceName: aRetrievedService.serviceName,
+  version: aRetrievedService.version
+};
+
+const servicesByScopeBlobId = "blob_id" as NonEmptyString;
+describe("GetVisibleServicesByScopeHandler", () => {
+  it("should get all visible services grouped by scope", async () => {
+    const blobStorageMock = {
+      getBlobToText: jest.fn().mockImplementation((_, __, ___, cb) => {
+        cb(
+          undefined,
+          JSON.stringify({
+            LOCAL: [aVisibleService.serviceId],
+            NATIONAL: [aVisibleService.serviceId]
+          })
+        );
+      })
+    };
+    const getVisibleServicesHandler = GetVisibleServicesByScopeHandler(
+      blobStorageMock as any,
+      servicesByScopeBlobId
+    );
+    const response = await getVisibleServicesHandler();
+    response.apply(MockResponse());
+
+    await Promise.resolve(); // needed to let the response promise complete
+    expect(blobStorageMock.getBlobToText).toHaveBeenCalledWith(
+      VISIBLE_SERVICE_CONTAINER,
+      servicesByScopeBlobId,
+      {},
+      expect.any(Function)
+    );
```

Comment:
It would be useful to test against the expected response payload. It would be independent from the implementation and most importantly it would serve as documentation example. We didi it [here](https://github.com/pagopa/io-functions-bonus/blob/1fb8478dcf51c9cff86f9e72d6292614f31b5d45/GetAllBonusActivations/__tests__/handler.test.ts#L66)

Reply:
Fixed in [d7c968e](https://github.com/pagopa/io-functions-assets/pull/3/commits/d7c968e6623871f17676406f07b7f8fa571f8471)